### PR TITLE
esp32c3: enable esp32c3 hal support 

### PR DESCRIPTION
**Pull request by uLipe**
Adds the support for esp32c3 specific hal files, also organize the zephyr adapters in a per-platform fashion.

### CODE REVIEW

**Comment by uLipe** [comment]
@glaubermaroto, @sylvioalves , @shubhamkulkarni97 PTAL again.

File: zephyr/esp32c3/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    ../../components/bootloader_support/include
+    )
+
+  zephyr_link_libraries(
+  	gcc
+    -T${CMAKE_CURRENT_SOURCE_DIR}/src/linker/esp32c3.rom.alias.ld
+  	-T${CMAKE_CURRENT_SOURCE_DIR}/../../components/esp_rom/esp32c3/ld/esp32c3.rom.ld
+  	-T${CMAKE_CURRENT_SOURCE_DIR}/../../components/esp_rom/esp32c3/ld/esp32c3.rom.api.ld
+  	-T${CMAKE_CURRENT_SOURCE_DIR}/../../components/esp_rom/esp32c3/ld/esp32c3.rom.libgcc.ld
+  	-T${CMAKE_CURRENT_SOURCE_DIR}/../../components/esp32c3/ld/esp32c3.peripherals.ld
```
Comment (glaubermaroto) [inline zephyr/esp32c3/CMakeLists.txt:39]: Please, check the indentation here.

**Comment by glaubermaroto** [comment]
@uLipe one minor comment, overall LGTM

File: zephyr/esp32c3/CMakeLists.txt (hunk 1 above)
Comment (uLipe) [reply zephyr/esp32c3/CMakeLists.txt]: Thanks @glaubermaroto , fixed!

File: zephyr/esp32c3/sdkconfig.defaults [hunk 2]
```diff
@@ -0,0 +1 @@
+CONFIG_ESP32C3_REV_MIN=3
```
Comment (sylvioalves) [inline zephyr/esp32c3/sdkconfig.defaults:1]: missing newline

File: zephyr/esp32c3/src/linker/esp32c3.rom.alias.ld [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+PROVIDE ( esp_rom_uart_tx_one_char = uart_tx_one_char );
+PROVIDE ( esp_rom_uart_rx_one_char = uart_rx_one_char );
+PROVIDE ( esp_rom_uart_attach = uart_attach );
+PROVIDE ( esp_rom_uart_tx_wait_idle = uart_tx_wait_idle );
+PROVIDE ( esp_rom_intr_matrix_set = intr_matrix_set );
+PROVIDE ( esp_rom_ets_set_user_start = ets_set_user_start );
+PROVIDE ( esp_rom_ets_printf = ets_printf );
+PROVIDE ( esp_rom_ets_delay_us = ets_delay_us );
+PROVIDE ( esp_rom_cache_set_idrom_mmu_size = Cache_Set_IDROM_MMU_Size );
```
Comment (sylvioalves) [inline zephyr/esp32c3/src/linker/esp32c3.rom.alias.ld:14]: missing newline

**Comment by sylvioalves** [comment]
@uLipe Please, add necessary `esp32c3` toolchain at `tools/zephyr_tools.json` in order to allow soc bootloader build. When user hits  `west espressif install`, it will install ESP32C3 toolchain.

**Comment by uLipe** [comment]
@sylvioalves i updated the zephyr_tools.json as you pointed out, thanks for the observation.

PTAL again.

**Review by sylvioalves — APPROVED** [review]

**Pull request merged**